src/CodeGenerator/src/Generator/OperationGenerator.php

```diff
@@ -137,7 +139,8 @@ public function generate(Operation $operation): void
         }
         if (null !== $operation->getOutput()) {
             $resultClass = $this->resultGenerator->generate($operation);
-            if (null !== $operation->getPagination()) {
+            $isPaginateListShape = false === \in_array($operation->getName(), self::PAGINATE_STRUCTURE_SHAPE);
+            if (null !== $operation->getPagination() && $isPaginateListShape) {
```

Comment:
IMHO this is not the right way to fix it.

It looks like the pagination.json declare a pagination available for the method, but nothing provide information about the localisation to the lines.

I suggest to patch the `pagination.json` entry instead:

in the manifest.json file, adds the following code:

```diff
 "Athena": {
   "source": "https://raw.githubusercontent.com/aws/aws-sdk- php/${LATEST}/src/data/athena/2017-05-18/api-2.json",
   "documentation": "https://raw.githubusercontent.com/aws/aws-sdk-php/${LATEST}/src/data/athena/2017-05-18/docs-2.json",
   ...
   "methods": [
     "GetCalculationExecution",
     "GetCalculationExecutionStatus",
     ...
   ],
+ "patches": {
+    "pagination": [
+      {
+        "value": "ResultSet.Rows",
+        "op": "add",
+        "path": "/pagination/GetQueryResults/result_key"
+      }
+    ]
+  }
 }
```

Reply:
The generation of GetQueryResults using the patch work.
But now, I got 2 new error in the CI due to that. One could be related to `.cache`